src/transformStateWithClones.js

```diff
@@ -7,7 +7,30 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let stateClone = Object.assign({}, state);
+
+  actions.forEach(function (element) {
+    if (element.type === 'clear') {
+      stateClone = {};
+      stateHistory.push({ ...stateClone });
+    }
+
+    if (element.type === 'addProperties') {
+      stateClone = { ...stateClone, ...element.extraData };
+      stateHistory.push({ ...stateClone });
+    }
+
+    if (element.type === 'removeProperties') {
+      element.keysToRemove.forEach(function (key) {
+        delete stateClone[key];
+      });
+
+      stateHistory.push({ ...stateClone });
```

Comment:
Similar to the previous comments, avoid repeating the push operation and move it to the end of the loop cycle, outside of the conditional statements.